packages/atomic/src/components/facets/facet-common.ts

```diff
@@ -23,3 +30,68 @@ export interface FacetValueProps {
   part?: string;
   buttonRef?: (element?: HTMLButtonElement) => void;
 }
+
+function isCategoryFacetValueRequest(
+  value: AnyFacetValueRequest
+): value is CategoryFacetValueRequest {
+  return 'children' in value && Array.isArray(value.children);
+}
+
+function isFacetValueRequest(
+  value: AnyFacetValueRequest
+): value is FacetValueRequest | CategoryFacetValueRequest {
+  return (
+    'value' in value &&
+    typeof value.value === 'string' &&
+    !('children' in value)
+  );
+}
+
+function getSelectedCategoryFacetValueRequest(
+  ancestor: CategoryFacetValueRequest
```

Comment:
`ancestor` might be a bit confusing here, when it's recursively calling the method w/ the children. Simply value would be fine.